### PR TITLE
Remove senior roles

### PR DESCRIPTION
**Pull request by EmoGarbage404**
## About the PR

Senior roles are generally ineffective at what they were designed to do. Intended to be teaching roles for interns, they fall short due to a longstanding lack of communication and mechanical reinforcement. It's not a stretch to say that, with their drippier clothes and high role timers, they are basically just free slots reserved for people with high hour count. 

The roles in general don't really have a job distinction that justifies them being separate and should honestly just be folded back into the regular jobs for the department (which is what I did)

This PR removes all senior roles from maps and in the files, additionally adjusting the role slots to maintain the same number of jobs per department.

## Media

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

the migrations obliterated the spawn points and the job prototypes are gone so that needs to be fixed wherever

**Changelog**

:cl:
- remove: Removed Senior Researcher, Senior Engineer, Senior Physician, and Senior Officer.

### CODE REVIEW

**Comment by JoeHammad1844** [comment]
I think a simple name change to something like medical teacher would suffice to make the job purpose clearer

**Comment by OliverOtter** [comment]
Senior roles are good for roleplay as well, plus a lot of people enjoy teaching new people to the game/department. Maybe a rename would be nice like the comment above, I dont think they should be fully removed.

**Comment by 27alaing** [comment]
will their clothing still be available?

**Comment by EmoGarbage404** [comment]
> Senior roles are good for roleplay as well, plus a lot of people enjoy teaching new people to the game/department. Maybe a rename would be nice like the comment above, I dont think they should be fully removed.

Why do we need extra roles for something that doesn't require a role to begin with. No other roles exist for the simple sense of "roleplay."

All it does is add in extra spawners, prototypes, and logic that needs to be considered by mappers and coders.

**Comment by keronshb** [comment]
The plan moving forward is to also remove the intern role and then just use job timers to add a prefix to the title, as needed. 
I believe there also might be something added for "Senior" roles in terms of flavor text job titles but please do not hold me to this, it was just speculation.
Also the HoP exists, if you really want your role to say Senior or Teacher, you can always ask your dept lead AND the HoP to add that prefix to your ID card.

**Comment by mirrorcult** [comment]
> No other roles exist for the simple sense of "roleplay."

i dont really think we need senior roles but this is reductive imo and also just not really true, lawyers reporters etc exist though they have more obvious grounding

**Comment by EmoGarbage404** [comment]
> > No other roles exist for the simple sense of "roleplay."
> 
> i dont really think we need senior roles but this is reductive imo and also just not really true, lawyers reporters etc exist though they have more obvious grounding

Fair. But even in the context of those roles, I'd argue that they're place in the roleplay of the game is reinforced by mechanical means like access and items they are given.

**Comment by nikthechampiongr** [comment]
I've personally seen senior roles used a lot to teach new players. Heads will often delegate that task to them. This is especially true for security in my experience. They also serve a good roleplay role where for example in security they may be made "squad leaders", or similar which is a neat thing. 

It is quite unusual that they are just an extra role while being functionally identical to non-senior roles. Maybe instead of just having a limit on the number of seniors, they should also draw from the same pool. For example a station can have 5 secoffs and one of them can be a senior, instead of 5 secoffs and 1 senior.

**Comment by Emisse** [comment]
Ideally this stuff is tied to loadout and role timers. I dont really care if its removed right now but it should be when loadout is merged with unlockable role outfits/titles

**Comment by TurboTrackerss14** [comment]
Copy and pasting my responce from an almost 4 month old bugs-feedback thread.

Not a coder/project manager so, grain of salt. 

Senior roles are a shown clear indicator of repeated playtime and commitment to a role. 
They fill the gap between Head of (job) and normal staff. 
Heads of staff have more responsibilities than just training freshies, they are a member of command. Managing more generally than directly in person. 

Take it from the point of security, a cadet is to be trained by someone knowledgeable. While this job COULD be done by the HoS, there are more pressing matters for them. Sure a normal secoff  could train them but as most every cadet will say "the moment I have enough role time for secoff im not going cadet". Showing that there is a clear line in the sand for maybe "not entirely trained" staff to get on to the standard role. Senior roles allow a visible indicator of large playtime, not just enough to know about those things but also a willingness to teach. 
>however I have seen many people just take senior roles for the drip and to try to give normal secoffs orders 

The creation of senior roles also means that heads of staff can manage more than train. taking some of their workload and pushing it to someone they KNOW will be experienced. Think of how many times you see a first time CE ask "how do i rotate in construction menu" etc.

**Comment by JoeHammad1844** [comment]
> Why do we need extra roles for something that doesn't require a role to begin with. No other roles exist for the simple sense of "roleplay."

psychologist
librarian

**Comment by JoeHammad1844** [comment]
seniors are good because when an intern comes up to me asking how to heal people i just tell them to talk to the senior :trollface:

**Comment by Rainbeon** [comment]
> No other roles exist for the simple sense of "roleplay."

Librarian is a non-station specific job that is entirely roleplay-centric, and it requires a much greater stress on mappers as you actually have to map out a library room for them instead of just giving them a spawner and job on the job list, quite a few other roleplay-based jobs as well like Psychologist, Reporter, Lawyer...

**Comment by EmoGarbage404** [comment]
> Librarian is a non-station specific job that is entirely roleplay-centric, and it requires a much greater stress on mappers as you actually have to map out a library room for them instead of just giving them a spawner and job on the job list, quite a few other roleplay-based jobs as well like Psychologist, Reporter, Lawyer...

librarian is hardly a shining example of a well-design role. neither are station-specific wild cards. Considering these are meant to be in their completed state, they are significantly lacking.

if you read through the [original pr](https://github.com/space-wizards/space-station-14/pull/18181), i think it's fairly obvious that a lot of the concerns about seniors becoming mini-heads and extra job slots are valid in retrospect. Even back then, plenty of people felt that these didn't need to be their own roles. frankly, i don't know why it was merged in the first place.

**Comment by Emisse** [comment]
Merged to try it out, now we can merge to untry it out

**Comment by UbaserB** [comment]
Personally, i loved teaching newbies. It was fun, and people knew i was experienced if anyone needed help!

**Comment by JoeHammad1844** [comment]
the change also ensures that somebody competent is playing, ive seen a bunch of idiotic doctors

**Comment by Hmeister-real** [comment]
I agree with this completely, i think a lot of people use these roles as a reservation for their high playtime, and even if they don't there isn't much benefit to choosing this role other than a free position in high-pop or roleplay. If the clothes from these roles are kept in the game, then I feel like it is fine to remove them. And for the argument that they are good to new players, the people who teach new players with these roles can just, teach new players with the normal job. Once for example, i was an engi and 2 engi interns came up to me setting up Singulo and i spent almost the whole round teaching them how to correctly set up singulo, how to wear engi eva suits, activate helmets, turn on internals etc. All without needing to be a funky unneccesary role.

**Comment by VigersRay** [comment]
Туда их

**Comment by aleksandrtikh** [comment]
- Senior roles are not just RP, teaching people stuff is important
- They usually have expanded access compared to the normal roles - they can serve as replacements for atmos or chem when needed
- Linked to the previous pt. - they can also teach the advanced roles, which normal roles are not guaranteed to know. So the chemist will go on making meds for the crew/the goal, while the second, newbie chemist is getting taught.
- People *can* teach even without a role, but they won't always *will*. With a senior role there's an assumption, a designation that serves as a grease of sorts. And in my experience (which isn't worth much, to be fair), it works.

**Comment by Mirino97** [comment]
Hi, My name is Commander Sammy in this is my favorite PR in the Project.

**Comment by OtherSkittle** [comment]
I can't count the number of times a player has ignored me because they were too busy try-harding their job to even be aware that other people exist, let alone be willing and able to help.

The way I see it people in regular roles need not and should not be expected to help their coworkers learn the job. I'm not saying they shouldn't but sometimes people

- don't know how to teach,

- don't know the job well enough to be confident in teaching it,

- don't feel like spending time teaching someone else,

- don't have the time because the section is overrun by requests from everyone else.

A senior role should be a clear and obvious tell to new players that the person in this role is willing and able to teach them how their job works. Now I'm not saying that this is currently the case but removing senior roles altogether seems like an extreme reaction that will make the situation worse.

As for the reserved slots argument, I don't really understand what the problem here is? The way I see it people should get reserved slots for having a large amount of time in a specific job because at least this kind of guarantees that there can be some people in each section with enough time to be competent enough at the game to keep the station running. Whether or not this is what happens in reality is also not really relevant and is mostly an issue of time-gating roles but this is not what the pr is about.

If the problem is having too many people in a single section to the point where most of them (probably the intern players) won't be allowed to participate in the job they signed up for, then I have bad news because even if you remove senior roles then it still feels like there are too many roles per section for each station because of the intern roles. For example: If I'm latejoining and want to play science but all the scientist roles are taken up, I as the player have zero incentive to not just pick the intern role and roleplay as an experienced scientist anyway, just so I can play in the section I want. I don't know if this is supposed to be intended but if it is then it seems kind of stupid that intern roles can be stolen from new players by experienced players wanting to play in a specific section.


> I think if the idea is to have just for example the scientist role but for play time to add a prefix to the role name dependant on hours in role. So only 1 hour and you would get the prefix intern, if you have 100 hours you get the prefix senior etc. This could be expanded to include other cool prefix words added to role names without the need for specific jobs for niche roles.

I agree with this and I would like to add that senior prefix should be optional with the caveat that the prefix is there to advertise the player's willingness to teach. People should also be punished for abusing senior roles/prefix (if they refuse to teach people or just use it to flex their hour count).

Intern prefix should also not be locked out of players who have enough hours because let's be honest we all already know time is a useless metric for job knowledge and people should still be allowed to advertise their inexperience in the chance that they do not feel confident enough in the job.

tl;dr: senior role is there to teach so regular roles dont have to, maybe roleban people that abuse senior roles to flex hours and refuse to teach interns.

**Comment by GJAAMER** [comment]
The idea couldn't be worse. I'm like a person who plays on MRP and HRP servers, I can say that these roles fit VERY well into the gameplay. These roles really help newbies, both in mastering the game mechanics and in the RP process.
Fuck, and in general: it’s easy to delete, better try to come up with something new and cool. STOP CASTRATING GAMEPLAY

**Comment by dffdff2423** [comment]
I am not going write as lengthy post as I intended because most of the points that I wanted to make have already been made.  As an MRP/HRP player the roles add a lot to the RP and regularly help new players. Quite a few of my most memorable RP moments where with senior roles (for instance I spent over 40 minutes one night learning about PPE as a stupid kobald). 

Adding titles based off playtime would not help because in that case (at least on MRP and HRP) for multiple reasons. First, it will most frequently lead to $DEPARTMENT being filled by entirely by whatever the highest title is (MRP/HRP players tend to have way too much playtime).  But more worryingly the converse could also happen where the department is filled entirely by interns likely resulting in the head giving up and saying "just figure it out yourself".  My other main issue would be that if titles are tied strictly to playtime it would be impossible to RP as a "newcomer". Another issue with having multiple people with "tutor titles" could lead to confusion about "who to ask". Limiting it to one role helps a ton. Finally, the additional access and exclusive clothes they get clearly marks them as well... a senior. A new player is unlikely to shift+click then look at id (or more likely right click then navigate through the long menu) to see a player's title and may not even know how to look at the manifest (I did not for about my first 20 hours).

Finally, the senior roles are great as an "I have a ton of playtime and I am willing to teach flag". When I was trying to learn the game before senior roles where added (on LRP) I would often end in the situation where nobody would want to teach me (after asking over radio like 3-4 times). I was able to almost fully unlock engi without knowing how to build walls or hack doors (until I gave up and looked up tutorials online). After seniors where added I did not have an equivalent experience.

Another possibility is that they get removed from LRP but not MRP if players are not playing them appropriately. I do not understand why we insist on having strict parity between MRP and LRP when they are very different environments. I feel that MRP is often neglected on decisions such as these.

At least keep them around until a proper title system is made (that hopefully addresses the above flaws).

**Comment by metalgearsloth** [comment]
> Another possibility is that they get removed from LRP but not MRP if players are not playing them appropriately. I do not understand why we insist on having strict parity between MRP and LRP when they are very different environments. I feel that MRP is often neglected on decisions such as these.

Because it doubles maintainer workload maintaining 2 entirely different games which we are not inclined to do at this time.

There's no harm in just trying out disabling it and bringing the drip back when loadouts are ready.

**Comment by dffdff2423** [comment]
> > Another possibility is that they get removed from LRP but not MRP if players are not playing them appropriately. I do not understand why we insist on having strict parity between MRP and LRP when they are very different environments. I feel that MRP is often neglected on decisions such as these.
> 
> Because it doubles maintainer workload maintaining 2 entirely different games which we are not inclined to do at this time. We make the game for LRP and MRP at the same time, not one or the other.

I should not have included that in my original post. It is very much something that is off topic for this discussion and I probably should not be even bringing it up as someone who has never maintained an open source codebase. I will edit my post to remove it.

**Comment by metalgearsloth** [comment]
Coming from ss13 over-specialising departments is bad and this crept too far into that direction. It's better to have 6 MDs rather than 3 MDs, a paramed, a senior role, and a geneticist. If no one takes paramed for example someone can just take it without having to worry about changing their access / gear to fit the job. Science also for example has many sub-departments but you only ever roll as a scientist, not a toxins specialist or xenobiologist.

If people want to teach others there's 0 RP reason for them to not just don a uniform and offer themselves as available.

**Comment by Hmeister-fake** [comment]
>Because it doubles maintainer workload maintaining 2 entirely different games which we are not inclined to do at this time. We make the game for LRP and MRP at the same time, not one or the other.

Isn't the whole point of wizden to be the experimental testing grounds server as they are the upstream for all the less extreme downstream servers? Like how Wizden has gotten all the big inventory tweaks, so that downstream can adjust in its own pace, such as delta-v and that new HRP one added recently. The lrp and mrp servers should never be out of sync with eachother because that ruins the whole point of why wizden exists.

**Comment by CMDR-Piboy314** [comment]
> This also applies to teaching roles where it also doubles as an exclusive slot for someone who has high playtime.

And does this apply to interns? Being an intern is an indication that you need to be taught and being a senior is an indication that you are there to teach.

**Comment by Acenius** [comment]
oh boy i sure am glad that now that the community has voiced its opinions on this public PR about a change in their community lead game everyone will stop arguing and the change will be reverted with slight changes that improve everyone's quality of life :godo:

anyways if someone abuses the senior role by refusing to teach that is a player issue not an issue with the role itself, as checking "senior ______" is like driving past a "vehicles on this premise are viable to be searched" sign, you've agreed to be the teaching role

**Comment by JoeHammad1844** [comment]
it actually isn't that obvious to people who haven't read the original PR that it's a teaching role, nobody reads the job descriptions and the name doesn't have any clues. we also cant police how people do their jobs unless we move to MRP

**Comment by OtherSkittle** [comment]
> it actually isn't that obvious to people who haven't read the original PR that it's a teaching role, nobody reads the job descriptions and the name doesn't have any clues. we also cant police how people do their jobs unless we move to MRP

If you put in the dozens of hours of gameplay required to unlock senior roles in the first place then it's probably on you to know what the single purpose of the role is.

As for new players I guess it isn't immediately obvious who you should turn to when learning the game and most videos about the subject also don't really mention senior roles. Personally when I started out I read what all the jobs do even if I already knew most of them (and frankly I would assume the type of person that is actually interested in being taught how the job works in a meaningful way and isn't just there to greytide would do the same). That being said a truly new player would probably also get overwhelmed by the amount of jobs and might not take note of senior roles so I'm not sure about this.

If senior roles can only be policed on MRP then new players need to have access to it by default otherwise senior roles should probably not even be bothered with.

**Comment by Vonsant** [comment]
So, looking back a year after removing the senior roles, was it a good decision? 🤔

**Comment by lzk228** [comment]
definitely yes

**Comment by LittleNorthStar** [comment]
Yeah I agree removing them was for the best, and I really like the new implementation as just a loadout thing it's much more elegant

**Comment by lzk228** [comment]
i want to attach #22462 pr just sso people would see how senior roles should to be implemented

**Comment by Vonsant** [comment]
> i want to attach #22462 issue just sso people could see how senior roles should to be implemented

I remember that. But it was also a year ago, and unfortunately, nothing has changed since then.

**Comment by Piras314** [comment]
I still do not like the removal of senior roles to this day :godmode:
